src/components/TextWithModal/index.tsx

```diff
@@ -25,6 +33,7 @@ export type TextWithModalProps = {
   onWhiteBackground?: boolean;
   clickOnText?: boolean;
   withGrayIcon?: boolean;
+  additionalIcon?: (props: AdditionalItemProps) => JSX.Element;
```

Comment:
Using render props - since we are passing down props back up to determine how to render the icon.
Seemed to be most suitable to avoid large refactors.

Open to other suggestions.